scheduler/src/cook/kubernetes/api.clj

```diff
@@ -82,16 +82,23 @@
   (when-not (synthetic-pod? pod-name)
     pod-name))
 
-(defn assoc-uuids
-  "Returns event-map with added fields :job-uuid (based on pod-name) and :instance-uuid (if pod is not synthetic)"
+(defn assoc-fields
+  "Returns event-map with added fields :instance-uuid (if pod is not synthetic), job-name, job-uuid, pool, and user."
   [event-map pod-name]
   (let [instance-uuid (pod-name->instance-uuid pod-name)
         job-uuid (or (pod-name->job-uuid pod-name)
-                     (cached-queries/instance-uuid->job-uuid-cache-lookup instance-uuid))]
+                     (cached-queries/instance-uuid->job-uuid-cache-lookup instance-uuid))
+        job-map (cached-queries/job-uuid->job-map-cache-lookup job-uuid)
+        job-name (:job/name job-map)
+        pool (:pool/name (:job/pool job-map))
+        user (:job/user job-map)]
```

Comment:
I think you can destructure these from `job-map`

Reply:
gotcha-- Does this look good: 
`{job-name :job/name user :job/user {pool-name :pool/name} :job/pool} job-map`

I also tried:
```
{:keys [job/name job/pool job/user]} jobmap
{:keys [pool/name]} pool
```
But then `name` is equal to `pool/name` and not `job/name`. And when I try to do something like `(prn job/name)`, I get a "No such namespace" exception.

If there's a better Clojure way to do this please let me know :).

Reply:
Your first proposal above looks good, except you can do it on the line where `job-map` is currently defined, and then you might not even need `job-map`, e.g.:

```
{job-name :job/name user :job/user {pool-name :pool/name} :job/pool} (cached-queries/job-uuid->job-map-cache-lookup job-uuid)
```